lib/diagnostic.js

```diff
@@ -59,14 +61,20 @@ return array;
 // If the argument obj is unset, create a new object
 // The function you write should return
 // the object passed in as an argument or the object created
+function Obj(propertyName, method) {
+  this[propertyName] = method;
+}
 const addMethod = function addMethod(propertyName, method, obj) {
+
 if (obj) {
-  this.propertyName = method;
+  obj[propertyName] = method;
 } else {
-  obj = {
-    propertyName: method
-  };
+  let obj = new Obj(propertyName, method);
+  return obj;
+
 }
+
+return obj;
 };
```

Comment:
BTW, when the question prompt says "your function should return..." that means your function needs to have a `return` statement at the end.